python/src/interpreter.cc

```diff
@@ -95,13 +88,15 @@ template <typename DType> class AtomicRMWOpBase : public AtomicOp {
 protected:
   void applyAt(void *loc, size_t i) override final {
     if (mask[i]) {
+      std::atomic<DType> *atomic_ptr = static_cast<std::atomic<DType> *>(loc);
```

Comment:
While this usually works in practice, it's also undefined behavior which isn't great. C++20 has `std::atomic_ref` which would fix this issue but we don't compile with C++20 yet.